python_modules/dagster/dagster/core/types/dagster_type.py

```diff
@@ -825,6 +827,8 @@ def resolve_dagster_type(dagster_type: object) -> DagsterType:
     if is_typing_type(dagster_type):
         dagster_type = transform_typing_type(dagster_type)
 
+    if get_origin(dagster_type) == Output:
+        dagster_type = get_args(dagster_type)[0]
```

Comment:
Not 100% confident in my reasoning, but shouldn't the `get_origin` section precede the `is_typing_type` call and subsequent conversion? If you use `Output[List[str]]`, in the current state the `List[str]` that you extract isn't going to get run through the typing type > dagster type conversion. Your test isn't catching this because it uses simple types (`str`/`int`).

Can you update the test to use a complex type like `List[str]` and see if it works?

Reply:
Ah that's a great catch. Will re-order